Get() bug if expected Object is not an Object

Given this Get() test case, 
```
        {
                desc:    `handle non-object and skip`,
                json:    `{"a":"d","c":{"c":[1,2]}} }`,
                path:    []string{"a", "c"},
                isFound: false,
        },
```
We get this:
```
--- FAIL: TestGet (0.00s)
        parser_test.go:1022: Get() test 'handle non-object and skip' isFound mismatch: expected false, obtained true
```

--- a/parser.go
+++ b/parser.go
@@ -254,6 +254,13 @@ func searchKeys(data []byte, keys ...string) int {
 					// If we found all keys in path
 					if keyLevel == lk {
 						return i + 1
+					} else {
+						// If there are more keys in the path, confirm the next
+						// token is the start of an array or an object
+						nextValOffset := nextToken(data[i+1:])
+						if data[i+1+nextValOffset] != '{' && data[i+1+nextValOffset] != '[' {
+							return -1
+						}
 					}
 				}
 			} else {

--- a/parser_test.go
+++ b/parser_test.go
@@ -519,6 +519,12 @@ var getTests = []GetTest{
 		data:    `{"c":"d" }`,
 	},
 	{
+		desc:    `handle non-object and skip`,
+		json:    `{"a":"d","c":{"c":[1,2]}} }`,
+		path:    []string{"a", "c"},
+		isFound: false,
+	},
+	{
 		desc:    `empty path`,
 		json:    `{"c":"d" }`,
 		path:    []string{},

